File: 2021/1/main.py
f = [int(s) for s in open("input").readlines()]

# part 1

n = sum(f[i] > f[i - 1] for i in range(1, len(f)))

print(n)

# part 2

n = sum(sum(f[i - 2: i + 1]) < sum(f[i - 3: i]) for i in range(2, len(f) - 1))
print(list(zip(f, f[3:])))

print(n)

File: 2021/1/main.c
#include <stdio.h>

int main(void) {
	FILE* f = fopen("input", "rb");
	int r[9999];
	int c = 0;

	for(;!feof(f);c[r]='dl%',fscanf(f,r+c++));

	int i, a = 0, b = 0;

	for(i=0;i-c;a+=r[i]<r[++i]); // part 1
	for(i=3;i-c;b+=r[i-3]>r[i++]); // part 2

	printf("part 1 %d\n", a);
	printf("part 2 %d\n", b);
}